mysql/datadog_checks/mysql/mysql.py

```diff
@@ -812,12 +819,14 @@ def _get_server_pid(self, db):
 
         # If pid has not been found, read it from ps
         if pid is None and PSUTIL_AVAILABLE:
-            try:
-                for proc in psutil.process_iter():
+            for proc in psutil.process_iter():
+                try:
                     if proc.name() == "mysqld":
                         pid = proc.pid
-            except Exception:
-                self.log.exception("Error while fetching mysql pid from psutil")
+                except (psutil.AccessDenied, psutil.ZombieProcess, psutil.NoSuchProcess):
+                    continue
```

Comment:
Can we debug log the exception here to help if we run into issues?

Reply:
Not sure it'd help, if the codepath gets there it means a process died during the iteration. If that process was exactly mysql, the pid would be `None` anyways so in terms of responsibilities of the function we're good. If that wasn't mysql, we don't care at all.